kafka-client/src/main/java/dev/responsive/db/CassandraClient.java

```diff
@@ -145,19 +150,59 @@ public OptionalInt numPartitions(final String tableName) {
     return numPartitions == 0 ? OptionalInt.empty() : OptionalInt.of(numPartitions);
   }
 
-  public RemoteKeyValueSchema kvSchema(final KVSchema schemaType) {
-    switch (schemaType) {
-      case KEY_VALUE: return kvSchema;
-      case FACT:      return factSchema;
-      default:        throw new IllegalArgumentException(schemaType.name());
+  public void shutdown() {
+    executor.shutdown();
+    session.close();
+  }
+
+  public RemoteKeyValueSchema prepareKVTableSchema(final ResponsiveKeyValueParams params)
+      throws TimeoutException, InterruptedException {
+
+    final RemoteKeyValueSchema schema;
+    switch (params.schemaType()) {
+      case KEY_VALUE:
+        schema = kvSchema;
+        break;
+      case FACT:
+        schema = factSchema;
+        break;
+      default:
+        throw new IllegalArgumentException(params.schemaType().name());
     }
+
+    schema.create(params.name().cassandraName(), params.timeToLive());
+    awaitTableAndPrepareSchema(params.name().cassandraName(), schema);
+
+    return schema;
   }
 
-  public RemoteWindowedSchema windowedSchema(final WindowSchema schemaType) {
-    switch (schemaType) {
-      case WINDOW:    return windowedSchema;
-      case STREAM:    throw new UnsupportedOperationException("Not yet implemented");
-      default:        throw new IllegalArgumentException(schemaType.name());
+  public RemoteWindowedSchema prepareWindowedTableSchema(final ResponsiveWindowParams params)
+      throws TimeoutException, InterruptedException {
+
+    final RemoteWindowedSchema schema;
+    switch (params.schemaType()) {
+      case WINDOW:
+        schema = windowedSchema;
+        break;
+      case STREAM:
+        throw new UnsupportedOperationException("Not yet implemented");
+      default:
+        throw new IllegalArgumentException(params.schemaType().name());
     }
+
+    schema.create(params.name().cassandraName(), Optional.empty());
+    awaitTableAndPrepareSchema(params.name().cassandraName(), schema);
+
+    return schema;
+  }
+
+  private void awaitTableAndPrepareSchema(
```

Comment:
basically moved all the table creation and schema prep to the client so we don't accidentally update eg the KV store but not the window store

Eventually I'll probably do a larger refactor to pull everything else shared in the stores to a common superclass, but I'm waiting to finish the window store stuff first